### ISSUE TITLE
[TASK] Revert affinity for Longhorn user deployed components

### ISSUE THREAD

**Issue by innobead** [longhorn/longhorn#5191]
@sjoukedv added affinity support for longhorn manager-based related component deployment. This can be applied to other user-deployed components as well to have a consistent scheduling mechanism. 

https://github.com/longhorn/longhorn/pull/5168

The only thing that needs to pay attention to is whether this change would potentially impact Longhorn deployment on a mixed Linux/Windows cluster. https://github.com/longhorn/longhorn/blob/76eaa3d3c17f2711855bf9594f4db5554dc242fc/chart/values.yaml#L4-L22 needs to be respected.

**Comment by ejweber** [comment]
@PhanLe1010 and I were discussing https://github.com/longhorn/longhorn/pull/5168 and have a concern:

Since we can now use node affinity to control which nodes longhorn-manager can deploy to (instead of just node selector), but system-managed components only respect node selector, there is the potential for some confusing behavior. If I use node affinity to express a complex requirement (e.g., run on nodes that ARE labeled key1:value1 and NOT labeled key1:value2), but cannot express that complex requirement with the setting `system-managed-components-node-selector`, there will be some nodes where `longhorn-manager` cannot run but pods like `longhorn-csi-plugin` do.

TLDR, it may be necessary to create a new `system-managed-components-node-affinity` if we want to enable users to use node affinity properly.

**Comment by ejweber** [comment]
Moving this to Review so I can start filling out testing steps. However, it can't be completed until we resolve the comment above. Please let us know your thoughts @innobead.

**Comment by innobead** [comment]
@ejweber exactly, we should make the behavior the same as what we support for node selector and taint toleration. 

Either fix the current issue with limitation without supporting system managed components which will be improved later, or do all things together.

If unable to achieve this in 1.5.0, let's move it to 1.6.0 instead.

**Comment by weizhe0422** [comment]
Test steps:
**Install within manifest**
  1. Install Longhorn with command: `kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/v1.5.0-rc1/deploy/longhorn.yaml`
  2. Verify the Longhorn is installed properly
     ![image](https://github.com/longhorn/longhorn/assets/17548901/b5c48c94-00de-4281-89a5-1e415a0b626a)
  3. Check volume could be created and attached
     ![image](https://github.com/longhorn/longhorn/assets/17548901/3bd3a0cc-064e-4032-b392-e64fa36e7ed4)
  4. The system can also be successfully uninstalled
     ![image](https://github.com/longhorn/longhorn/assets/17548901/6fc1413e-cd25-43a9-a4bb-1d44b2de655b)
